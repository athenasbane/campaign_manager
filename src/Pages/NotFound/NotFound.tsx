import Typography from "@mui/material/Typography";

export default function NotFound() {
  return (
    <>
      <Typography variant="h1">404 Page Not Found</Typography>
    </>
  );
}
