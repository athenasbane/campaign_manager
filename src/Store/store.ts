import { combineReducers, configureStore } from "@reduxjs/toolkit";
import modalReducer from "./slices/modals";
import { contentfulApi } from "./slices/backend";
import activeMissionReducer from "./slices/activeMission";
import layoutReducer from "./slices/layout";

const rootReducer = combineReducers({
  activeMission: activeMissionReducer,
  modals: modalReducer,
  layout: layoutReducer,
  [contentfulApi.reducerPath]: contentfulApi.reducer,
});

export function setupStore(preloadedState?: Partial<RootState>) {
  return configureStore({
    reducer: rootReducer,
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware({ serializableCheck: false }).concat(
        contentfulApi.middleware
      ),
    preloadedState,
  });
}
const store = setupStore();

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof rootReducer>;

export type AppStore = ReturnType<typeof setupStore>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;

export default store;
