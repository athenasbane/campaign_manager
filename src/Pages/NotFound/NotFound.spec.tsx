import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import NotFound from "Pages/NotFound/NotFound";
import { Provider } from "react-redux";
import store from "Store/store";

describe("Page - NotFound", () => {
  it("should match snapshot", () => {
    const { container } = render(
      <Provider store={store}>
        <NotFound />
      </Provider>,
      { wrapper: BrowserRouter }
    );

    expect(container).toMatchSnapshot();
  });
});
