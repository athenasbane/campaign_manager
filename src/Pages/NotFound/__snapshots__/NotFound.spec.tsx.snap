// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page - NotFound should match snapshot 1`] = `
<div>
  <h1
    class="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
  >
    404 Page Not Found
  </h1>
</div>
`;
